.github/actions/action.yml

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020-2022 F4PGA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: "Run tests"
+
+inputs:
+  plugin:
+    description: "Name of the plugin to run the tests for"
+    required: true
+  yosys-version:
+    description: "Which yosys version to build against ('conda' or branch name)"
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+
+    - uses: actions/setup-python@v2
+
+    - name: Install
+      shell: bash
+      run: |
```

Comment:
I suggest building a container in one step and reusing it for all the jobs, instead of installing all the deps in each jobs. Nevertheless, that can be done in a follow-up PR.